haystack/document_stores/memory.py

```diff
@@ -50,7 +50,7 @@ def __init__(
         scoring_batch_size: int = 500000,
         devices: Optional[List[Union[str, torch.device]]] = None,
         use_bm25: bool = False,
-        bm25_tokenization_regex: str = r"(?u)\b\w\w+\b",
+        bm25_tokenize_with: Optional[Union[str, Callable[[str], List[str]]]] = r"(?u)\b\w\w+\b",
```

Comment:
Why make it `Optional`? 🤔 
Same question for `bm25_tokenize_with`.

Reply:
Will remove that.